rl4co/envs/routing/tsp.py

```diff
@@ -161,6 +174,50 @@ def check_solution_validity(td: TensorDict, actions: torch.Tensor):
             == actions.data.sort(1)[0]
         ).all(), "Invalid tour"
 
+    @staticmethod
+    def improve_solution(td: TensorDict, actions: torch.Tensor, **kwargs) -> torch.Tensor:
+        """Improve the solution using local search"""
+        assert pyvrp is not None, (
+            "`pyvrp` not found. Please install pyvrp using instructions from "
+            "https://github.com/PyVRP/PyVRP or just use pip install pyvrp."
+        )
+        
+        rng = RandomNumberGenerator(kwargs.get("seed") or 0)
+        
+        action_list = actions.cpu().detach().numpy().tolist()
+        scaled_locs = scale_and_truncate_to_decimals(td["locs"].cpu().detach().numpy(), 7)
```

Comment:
Support for `scale_and_truncate_to_decimals` will also be dropped in the next release. I suggest to take copy-paste the implementation.

```python
def scale_and_truncate_to_decimals(vals: np.ndarray, decimals: int = 0):
    return (vals * (10**decimals)).astype(int)
```